playground/frontend/playground_components/lib/src/controllers/snippet_editing_controller.dart

```diff
@@ -156,19 +85,32 @@ class SnippetEditingController extends ChangeNotifier {
   bool get isChanged => _isChanged;
 
   void _updateIsChanged() {
-    _isChanged = _isCodeChanged() || _arePipelineOptionsChanged();
+    _isChanged = _calculateIsChanged();
   }
 
-  bool _isCodeChanged() {
-    return _selectedExample?.source != codeController.fullText;
+  bool _calculateIsChanged() {
+    for (final controller in fileControllers) {
```

Comment:
Rename `controller` to `fileController`.

Reply:
Local unambiguous variables are better short.